src/main/java/no/nav/fo/veilarbregistrering/arbeidssoker/ArbeidssokerService.java

```diff
@@ -43,6 +45,47 @@ public void behandle(EndretFormidlingsgruppeCommand endretFormidlingsgruppeComma
         arbeidssokerRepository.lagre(endretFormidlingsgruppeCommand);
     }
 
+    public List<Arbeidssokerperiode> hentArbeidssokerperioder(Bruker bruker, Periode forespurtPeriode) {
+        // Hent arbeidssøkerperioder lokalt for alle fnr fordi det er billig
+        Arbeidssokerperioder arbeidssokerperioderLokalt = hentAlleArbeidssokerperioderLokalt(bruker).sorterOgPopulerTilDato();
+
+        if (arbeidssokerperioderLokalt.dekkerHele(forespurtPeriode) && brukLokalCache()) {
+            List<Arbeidssokerperiode> overlappendeArbeidssokerperioderLokalt = arbeidssokerperioderLokalt.overlapperMed(forespurtPeriode);
+            Metrics.reportTags(Metrics.Event.HENT_ARBEIDSSOKERPERIODER_KILDE, Kilde.LOKAL);
+            LOG.info(String.format("Arbeidssokerperiodene fra egen database dekker hele perioden, og returneres: %s", overlappendeArbeidssokerperioderLokalt));
+            return overlappendeArbeidssokerperioderLokalt;
+        }
+
+
+        // Hvis ikke dekkende, sjekk ORDS for gjeldende fnr
+        Arbeidssokerperioder arbeidssokerperioderORDS = formidlingsgruppeGateway.finnArbeissokerperioder(bruker.getGjeldendeFoedselsnummer(), forespurtPeriode);
+
+        if (arbeidssokerperioderORDS.dekkerHele(forespurtPeriode)) {
+            List<Arbeidssokerperiode> overlappendeArbeidssokerperioderORDS = arbeidssokerperioderORDS.overlapperMed(forespurtPeriode);
+            Metrics.reportTags(Metrics.Event.HENT_ARBEIDSSOKERPERIODER_KILDE, Kilde.ORDS);
+            LOG.info(String.format("Returnerer arbeidssokerperioder fra Arena sin ORDS-tjeneste: %s", overlappendeArbeidssokerperioderORDS));
+            return overlappendeArbeidssokerperioderORDS;
+        }
+
+        // Hvis ikke dekkende, sjekk ORDS for alle fnr
+        List<Arbeidssokerperioder> historiskeArbeidssokerperioderORDS = bruker.getHistoriskeFoedselsnummer().stream()
+                .map(foedselsnummer -> formidlingsgruppeGateway.finnArbeissokerperioder(foedselsnummer, forespurtPeriode))
+                .collect(Collectors.toList());
+
+        return arbeidssokerperioderORDS.slaaSammenMed(historiskeArbeidssokerperioderORDS).overlapperMed(forespurtPeriode);
+    }
+
+    Arbeidssokerperioder hentAlleArbeidssokerperioderLokalt(Bruker bruker) {
+        Arbeidssokerperioder arbeidssokerperioder = arbeidssokerRepository
+                .finnFormidlingsgrupper(bruker.getGjeldendeFoedselsnummer());
+
+        List<Arbeidssokerperioder> historiskeArbeidssokerperioder = bruker.getHistoriskeFoedselsnummer().stream()
```

Comment:
Jeg synes vi skal dytte mer av logiske til Repository ift. å håndtere flere fnr/dnr. Repo burde bare kunne ta inn Bruker, og selv avgjøre om den skal hente på bakgrunn av én eller alle. Evt. ha to metoder om vi trenger det.